generic_tracers/cobalt_send_diag.F90

```diff
@@ -0,0 +1,2622 @@
+!        model_time, rmask = grid_tmask(:,:,1),&
+!        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+!==============================================================================================================
+! JGJ 2016/08/08 CMIP6 OcnBgchem day: Marine Biogeochemical daily fields
+! chlos and phycos - in Omon and Oday
+!==============================================================================================================
+! 2016/08/15 JGJ: 100m integrals w/o CMOR conversion
+
+        used = g_send_data(cobalt%id_f_dic_int_100,  cobalt%f_dic_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_f_din_int_100,  cobalt%f_din_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_f_po4_int_100,  cobalt%f_po4_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_f_fed_int_100,  cobalt%f_fed_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_f_sio4_int_100,  cobalt%f_sio4_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_f_alk_int_100,  cobalt%f_alk_int_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jdic_100,  cobalt%jdic_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jdin_100,  cobalt%jdin_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jpo4_100,  cobalt%jpo4_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jfed_100,  cobalt%jfed_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jsio4_100,  cobalt%jsio4_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+        used = g_send_data(cobalt%id_jalk_100,  cobalt%jalk_100,   &
+        model_time, rmask = grid_tmask(:,:,1),&
+        is_in=isc, js_in=jsc, ie_in=iec, je_in=jec)
+
+!==============================================================================================================
+
+    end subroutine cobalt_send_diagnostics   
+
+
+    !> subroutine that handles send_diag calls for bottom fluxes      
+    subroutine cobalt_send_diagnostics_bottom(tracer_list,model_time,grid_tmask,temp_field,dt,&
```

Comment:
This routine does more than just send the bottom diagnostics.  It also updates the bottom reservoirs and sets the bottom fluxes for later application as a bottom boundary condition.  I think the latter function may be more important than its diagnostic role (and was captured by the "update_from_bottom" naming.  

We may be able to separate the diagnostic function from "update_from_bottom", but I do wonder whether the timing of the bottom update relative to "update_from_source" may be a consideration here.  Ideally, the bottom flux save would be aligned with the time step during which it is applied in the tridiagonal solver.  We should be able to determine this once we have a flow diagram of generic tracer interactions with MOM6.

For now, maybe we should keep the generic_cobalt_update_from_bottom intact for this pull request and open and open issue on the bottom diagnostics that we can resolve later?

Reply:
Agreed. I have reverted these changes, and we will revisit the bottom flux issue later.